test/Middleware/AuthHttpBasicTest.php

```diff
@@ -25,72 +25,122 @@ protected function getMiddleware()
         return new AuthHttpBasic(
             $this->authDriver,
             $this->registry
+            
         );
     }
    
 
-    public function testNotAuthenticatedWithoutHeader()  //test ob user nicht authentifiziert
+    public function testNotAuthenticatedWithoutHeader()  
     {
         $username = 'testUser01';
-        
-        $this->authDriver->method('authenticate')->willReturn(false); //not authenticated
+
+        //not authenticated
+        $this->authDriver->method('authenticate')->willReturn(false); 
+        //new login request
         $this->registry->method('getAuth')->willReturn($username);
+        
         $request = $this->requestFactory->createServerRequest('GET', '/test');
         $middleware = $this->getMiddleware();
         $response = $middleware->process($request, $this->handler);
         
-        $noAuthHeader = $this->recentlyHandledRequest->getAttribute('NO_AUTH_HEADER'); //ohne authorization header
-     
-        $notAuth=$this->recentlyHandledRequest->getAttribute('HORDE_AUTHENTICATED_USER'); //nicht authorisiert->willreturn(false)
+        //no authorization header
+        $noAuthHeader = $this->recentlyHandledRequest->getAttribute('NO_AUTH_HEADER'); 
+        //NULL = notAuthenticated
+        $notAuth=$this->recentlyHandledRequest->getAttribute('HORDE_AUTHENTICATED_USER'); 
         
-        $this->assertNull($notAuth); //not authenticated
+        $this->assertNull($notAuth); 
         $this->assertEquals($username, $noAuthHeader); 
        
         
       
     }
     
-    public function testNotAuthenticatedWithHeader()  //test ob user nicht authentifiziert
+    public function testNotAuthenticatedWithHeader()  
     {
         $username = 'testUser01';
         
+        //not authenticated
         $this->authDriver->method('authenticate')->willReturn(false);
+
         $this->registry->method('getAuth')->willReturn($username);
-        $request = $this->requestFactory->createServerRequest('GET', '/test');
+        
+        
+        //with authorization header 
+        $request = $this->requestFactory->createServerRequest('GET', '/test')->withHeader('Authorization','BASIC YXNkZjphc2QxMjM=');
```

Comment:
It would probably better to create the base64 encoded string before this line to make the test it more readable:
```
$username = 'testUser01';
$password = 'testPw';
$authString = base64_encode(sprintf('%s:%s', $username, $password));
$request = $this->requestFactory->createServerRequest('GET', '/test')->withHeader('Authorization','BASIC ' . $authString);
```